### ISSUE TITLE
Consider sharing "hex digit to int" code instead of duplicating it

### ISSUE THREAD

**Issue by bzbarsky-apple** [project-chip/connectedhomeip#3769]
#### Problem
`HexDigitToInt` in `BleUUID.cpp` and `HexToInt` in `src/lib/mdns/DiscoveryManager.cpp` look like the same function.  

 #### Proposed Solution
Should probably be moved to a utility place of some sort.

**Comment by bzbarsky-apple** [comment]
@gjc13

**Comment by gjc13** [comment]
Ack

**Comment by Damian-Nordic** [comment]
@bzbarsky-apple I think @gjc13 fixed the issue, but didn't use the magic word "fix" in the PR, so the issue didn't get closed automatically ;).

**Comment by bzbarsky-apple** [comment]
No, this is not fixed.  In particular, the PR that all the noise is about above never got landed.

**Comment by cecille** [comment]
Some of this code has changed, but HexDigitToInt could probably use the Encoding::HexToBytes function
